pandas/core/arrays/datetimes.py

```diff
@@ -477,10 +477,20 @@ def _generate_range(
             arr = arr.astype("M8[ns]", copy=False)
             index = cls._simple_new(arr, freq=None, dtype=dtype)
 
-        if not left_closed and len(index) and index[0] == start:
-            index = index[1:]
-        if not right_closed and len(index) and index[-1] == end:
-            index = index[:-1]
+        # do not remove when one side is inclusive
+        # and removing would leave index empty
+        to_remove_any = not (
+            (left_inclusive or right_inclusive)
+            and len(index) == 1
```

Comment:
shouldn't this be `len(index)` ?

Reply:
mm yeah, changed that and changed the tests in test_date_range